src/main/java/net/sf/jabref/importer/fetcher/DOItoBibTeXFetcher.java

```diff
@@ -86,14 +85,18 @@ public JPanel getOptionsPanel() {
         return null;
     }
 
-    public BibEntry getEntryFromDOI(String doiStr, ParserResult parserResult) {
-        Objects.requireNonNull(parserResult);
+    public BibEntry getEntryFromDOI(String doiStr) {
+        return getEntryFromDOI(doiStr, null);
+    }
 
+    public BibEntry getEntryFromDOI(String doiStr, ParserResult parserResult) {
         DOI doi;
         try {
             doi = new DOI(doiStr);
         } catch (IllegalArgumentException e) {
-            parserResult.addWarning(Localization.lang("Invalid DOI: '%0'.", doiStr));
+            if (parserResult != null) {
+                parserResult.addWarning(Localization.lang("Invalid DOI: '%0'.", doiStr));
+            }
             return null;
```

Comment:
Could this be replaced with Optional<Bibentry> as a return type? We try to minimize the usage of null.